lib/rules/array.js

```diff
@@ -5,6 +5,16 @@
 module.exports = function ({ schema, messages }, path, context) {
 	const src = [];
 
+	let sanitized = false;
+	if (schema.convert === true) {
```

Comment:
I think we shouldn't have to wrap the `null` and `undefined` values because they are no "real" values.
The current implementation will wrap the `{ roles: null }` to `{ roles: [null] }`